perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -137,6 +144,69 @@ public static String join(Iterable<?> collection, String separator) {
 		return oBuilder.toString();
 	}
 
+	/**
+	 * Returns additionalUserExtSource according to the attrName
+	 *
+	 * @param attrName name of the attribute
+	 * @param subjectFromLoginSource map with the subject
+	 * @param sess perun session
+	 * @param perunBl PerunBl to get ExtSourceManagerBl
+	 * @return UserExtSource if exists, null if parameter attrName or subjectFromLoginSource is null or attrName doesn't start with ExtSourcesManagerImpl.USEREXTSOURCEMAPPING
+	 * @throws InternalErrorException
+	 */
+	public static UserExtSource getAdditionalUserExtSource(String attrName, Map<String, String> subjectFromLoginSource, PerunSession sess, PerunBl perunBl) throws InternalErrorException {
```

Comment:
Session is always first. You don't need to send perunBl to the method, you can find it in the perun session.

Reply:
I've managed to get it somehow but if there is a better way, please let me know.

Reply:
Resolved in personal discussion.